### ISSUE TITLE
Relationships need to be automatically created for a collection. 

### ISSUE THREAD

**Issue by dtarb** [hydroshare/hydroshare#1403]
When a resource is added to a collection the collection needs a relationship: dcterms:hasPart that points to the resource added to the collection.  (see http://purl.org/dc/terms/hasPart)

The resource being added to the collection should also have a relationship added dcterms:isPartOf.  Adding this term requires edit permissions on the resource, which the collection editor may not always have.  So, I suggest only creating this relationship automatically if permissions allow it, otherwise leaving it out, which would have the effect that the resource does not record collections it is part of.

**Comment by zhiyuli** [comment]
@dtarb @horsburgh @pkdash 
The bi-directional relationship between collections and resources are stored as ManyToManyFiled in django db. It is not stored under the "Relation" attribute of a resource.

Currently, user can see this relationship at two places:
1) Collection landing page, listing all contained resources
2) The "resourcemap.xml" file in collection bag, describing this relationship using the Dublin core terms "aggregates" and "isAggregatedBy". A "resourcemap.xml" sample file is here:
https://www.hydroshare.org/django_irods/download/d7e01ef89be14513bd24a7981bc7b134/data/contents/resourcemap.xml
It describes the relationship between a collection (fcaf8d5112f14f738819c818a34c9022) and its two contained resources (cdbb5ef9abbd4c4d8405c208d157a813 and 2163f9d3ad7f42ff81cca2dd8f8cbc41).
I remembered this format has been verified by @horsburgh.

Should we keep using the terms "aggregates" and "isAggregatedBy"? or change to use "hasPart" and "isPartOf".

Note that currently "aggregates", "isAggregatedBy" or "hasPart" are not options in the "Relation" attribute of a res. 
We only have the following options:
![image](https://cloud.githubusercontent.com/assets/10070599/17753994/a5351962-648f-11e6-9a98-ea622ca4695d.png)

Also I find I can create only one "IsPartOf" relation for each resource on landing page. When I tried adding a second "IsPartOf" relation, an error showed up:
![image](https://cloud.githubusercontent.com/assets/10070599/17754209/aa9d9248-6490-11e6-9277-587f161a382b.png)

**Comment by horsburgh** [comment]
@zhiyuli - there are a couple of issues here. 

1. I'm glad we got the resource map worked out. But, that's mainly for a computer to traverse the structure of a resource. I don't expect a lot of users to spend a lot of time viewing the resource maps. So, we should consider whether having the relationships there is enough.  For the resource map, we should keep the specific semantic terms "aggregates" and "isAggregatedBy" because those are OAI-ORE terms and are required by DataONE if we want to be compatible.

2. Given number 1, we really should have the relationships stored somewhere in the resource that is accessible to the user once the resource is downloaded. I think that since the resource is by definition a collection, the collected items should be listed in the science metadata for the resource.  For the collection resource, a list of "hasPart" relationships would be fine.  The fact that that term doesn't show up in the "Relation type" drop down just means that we need to add it.  For a collection, those relationships should be added automatically by HydroShare when a resource is added to the collection.

The error with the "isPartOf" term is likely because it was assumed that there would only be one relationship of that type per resource. Now that we have worked through issues with collections, it's apparent that a resource might be part of multiple collections.  So, this constraint should be lifted.

**Comment by zhiyuli** [comment]
@horsburgh 
OK, I think I got mixed up about Dublin core terms and OAI-ORE terms. 
So the Dublin core terms are used in resourcemetadata.xml; The OAI-ORE terms are used in resourcemap.xml. 
Can you please elaborate a bit more about the connection and difference between the two? thanks

**Comment by zhiyuli** [comment]
@dtarb @horsburgh @pkdash 
I listed four scenarios that relate to this issue, please let me know if I ignore anything.

1) Add a new resource to a collection:
   collection creates a "Has Part" Relation to this resource
   resource creates a "Part Of" Relation to this collection   

2) Remove a resource from a collection:
   collection removes "Has Part" Relation to this resource 
   resource removes "Part Of" Relation to this collection   

3) Delete a collection res:  
   each of the contained resources should remove the "Part Of" Relation to this collection

4) Delete a resource that is contained by one or more collections:
   those affected collections should remove "Has Part" Relation to this resource

Note: 
A) All of the above scenarios need to update bags
B) Since collection can hold collections, the above "resource" itself can be a collection. So when deleting a collection, this collection should be treated as a collection (see 3) as well as a resource (see 4).

**Comment by pkdash** [comment]
@zhiyuli The above actions related to relationship between resource and a collection seem correct to me. However, it may not be always possible to remove a relationship ('Has Part', 'Part Of') due to permission issue. That could create broken links in the xml document.

**Comment by horsburgh** [comment]
@zhiyuli and @pkdash - First, I suggest another major simplification to collections - unless there is a major driving use case, I think we should NOT allow collections to contain other collections. That should simplify some logic.

Then:

* A collection should point to all of it's collected resources using a "hasPart" relationship.
* A collected resource should point to the collection that it is a member of with a "partOf" relationship. A resource may be a member of many collections.

I've already suggested that we limit creation of collections to public OR published resources, which avoids the user specific permissions issues in viewing the "big bag." Another simplifying measure might be to limit users to creating collections of resources they have edit permissions on.  This would avoid the inability to remove relationships from resources because the user would have write permissions. There would be two remaining issues:

1. You can't edit the metadata of Published resources, so you wouldn't be able to add a "partOf" relationship to a published resource.
2. If a user adds a resource that they have edit permissions on to a collection and then their edit permissions are revoked, you would no longer be able to remove a "partOf" relationship from that resource if it is removed from the collection.

So, @pkdash is right, even with all of these simplifications, there would still be some instances where you wouldn't be able to remove relationships.  

To address this, we could make an even more drastic simplification and say that we are not going to add "partOf" relationships to collected resources at all. Then, there is only the potential for broken relationships in a collection resource if one of its members is deleted from HydroShare but not from the collection.  I'm not sure we can avoid this if we allow public (but unpublished and therefore mutable and deletable) resources in collections.

Finally - to avoid all of these issues, we could do what others have done and ONLY allow collections of published resources.  This may limit utility of collections, but seriously simplifies implementation.

**Comment by zhiyuli** [comment]
@dtarb  @horsburgh @pkdash 

A) **"NOT allowing collections to contain other collections"** will definitely simplify current implementation and several future works. Also I am thinking a combination usage of  the upcoming "folder" and "file type" features can meet similar purpose of "collection holds collections".
But it seems @ChristinaB group is already using "collection holds collections " to organize their resources. So it may require PIs to help coordinate this.

B) As for the permission issues, how about we just allow collection to exempt from some access control rules. Specifically, the "Has Part" and "Is Part Of" Relations will be added to or removed from all affected collections and resources automatically without checking any permission. Those actions can be considered as "system level actions", which are higher than "user level actions".  The only issue is when updating bags we need to set the value of "last change user". In this case we simply set it to "Admin".

**Comment by ChristinaB** [comment]
Regarding A) I am in the middle of getting NFIE summer projects as a Collection Use case onto HydroShare.  They have a Group, and they have a Collection of Projects for 2016.  In the current implementation, each project will be a Collection, where each group will load their project resources.   For projects with multiple students, such as NFIE innovators, I think Collections of Collections works great.  

If this all gets reworked and I need 26 people to reload everything in a different organization everyone will be frustrated.   If we think NFIE summer institute organization and publication of data is a good use case, I need to ask them first if they are willing to be HydroShare guinea pigs.

**Comment by horsburgh** [comment]
@ChristinaB - can you elaborate on the "collection of collections" part so I'm sure we understand it? Don't all the projects have multiple students? Why does this require a collection of collections?

**Comment by dtarb** [comment]
@zhiyuli @ChristinaB @horsburgh is seems like there is a bit of sorting out to do.  I've put this on the agenda for the call on Wednesday.  Please in preparation for the call organize and sort out the issues and use cases to support.

In the project Google Drive folder in System Design there is a folder for Collection Resource.  Right now it only contains a document on Collection permission rules.  It may be helpful to write out the use cases and functionality required a bit more specifically than in this document.  This does not need to be long.  See other examples in System design (e.g. FolderswithinResources).  If the three of you could start this prior to the call, then we would at least have the issues on the table to consider.

I'd also like to not be too drastic in changes to capability that seems to be working and perhaps meeting 80% of the need.  As I see it the issues on the table are:

1. Displaying collection relationships #1387 
2. Downloads #1304

As we contemplate changes to collections I want to be careful not to disrupt useful functionality we have just over these issues.

It is probably best to leave the discussion of downloads in #1304.  

With respect to displaying collection relationships, the first thing is to use the relationships metadata capability to store the relationships.  That is what I suggested in this issue.  Then once the relationships are stored, we can discuss whether they should be displayed for the user in the relationships tab or at the top as is being suggested in #1387.  When a user adds a resource to a collection, they are editing the collection resource, so adding the hasPart relationship should not be a problem.  Adding the isPartOf relationship to the target resource may be a problem if the user does not have permissions on that resource.  Some of the simplifying suggestions (only collections of public resources and no collections of collections) do not address this as the collection editor still may not have permission on the target public resource, and permissions for a collection resource are the same as for any other resource.  I see the following options for resolving this:  (1).  Where the collection editor does not have permissions, the target resource just does not get the isPartOf relationship.  This may not be that drastic a shortcoming.  (2) We determine as policy that the isPartOf relationship is metadata that is not subject to the access control rules.  It is removed from user control and only changed by the system in target resources, whenever a resource is added or removed from a collection.  I believe that Django already does this internally, we would just need to extend this to the display of relationships in the relationships tab and metadata xml file.  This may necessitate removing the ability for a user to directly create an isPartOf relationship and reserve this use for collections (or have separate terms for isPartOfCollection and isPartOf something else that a user can create).

**Comment by horsburgh** [comment]
@dtarb - I still think users should be required to have at least edit permissions on a resource to be able to add it to a collection.  As a HydroShare user, I don't personally feel like just anyone should be able to add my resources to a collection.  For instance, what would stop me from creating a new resource called "The top 10 worst HydroShare resources ever" and then adding a couple of your public resources to it.  You wouldn't be able to edit the collection to remove your resources. 

So in summary of this and my previous comments, I think we should implement the following restrictions on collections:

1. No collection resources within other collection resources (pending better description from Christina on potential use cases and whether we can support them in other ways)
2. Collections can be formed of only public and/or published resources (to alleviate permissions issues related to viewers of a collection resource)
3. Users must have at least edit permissions on a resource to add it to a collection (to alleviate permissions issues related to creating a collection resource)

**Comment by dtarb** [comment]
No one has permission to edit a published resource so points 2 and 3 above are in conflict and would preclude ever adding a published resource to a collection.

**Comment by horsburgh** [comment]
@dtarb - yes, we may still have to enable editing of relationships for published resources if we want to enable users to be able to add them to collections. Or - we just don't copy relationships to published resources. We have to choose. 

We can check with the access control team, but I don't think publishing a resource affects who has edit permissions. The resource becomes immutable but the access control rules remain intact don't they?

**Comment by dtarb** [comment]
OK.  So your suggestion is that they have to have edit permissions, but these are not used to edit the resource, only to evaluate whether the resource can be added to the collection.  Somewhat obscure, but could work.  We would have to check whether there is still capability to give someone edit permissions to enable this after a resource is published. I do worry that this is a bit of a roundabout way to solve a problem that has not occurred yet.  My concept of a collection is that it is just a list of resources.  I can make a list of resources and say anything I like about them.  I am not sure we want to restrict putting something that you have in my list if you do not want me to.

What do you think of my suggestion to determine as policy that the isPartOf relationship is metadata that is not subject to the access control rules.

**Comment by pkdash** [comment]
I assume this specific relationship is set by the system and not by the user directly. In that case it is not hard to by pass access rules. However, in case of a published resource is it okay to have an update date that's after the publish date?

**Comment by ChristinaB** [comment]
Collection of Collections: two examples of how it is on the verge of being used now. 
1. Nathan Chaney publishes his POLARIS soil database in Hydroshare as a Collection that holds geographic subsets of the databse. No one wants to download the entire thing for the CONUS, so as people ask for subsets, he adds them to his Collection.  That way, I can do a modeling project collection, and add his resource for my watershed from his POLARIS collection to my modeling project collection.  People who use my  model after me, can grab the entire collection, or just use the soils data.  This is the equal but opposite case to Jeff's '10 worst Hydroshare resources'  scenario. 

2. NFIE summer institute is one collection.  Every year has  5-10 projects, each project is a collection.  Thereby making the NFIE summer institute, a collection of collections.  Each collection in the NFIE collection is managed independently by 3-4 students, who both share their work with the NFIE collection,  build on it as a shared project collection with their summer friends, and have their own  independent resources which they developed.  

2.

**Comment by ChristinaB** [comment]
Whoops. 1. and 2. above.

**Comment by ChristinaB** [comment]
If these use cases can be met with other functions or steps, I just need to know what they are/will be so that I can explain to users (and myself) how to organize, share and build on work.

**Comment by horsburgh** [comment]
@ChristinaB - thanks for posting more details.  Some thoughts:

**Use Case 1**:  If you were to use a subset of the POLARIS data in your resource, that subset would be a resource in its own right. It would have to be a resource on its own to be part of the POLARIS collection and for you to reference it in your collection. Given this, you have a couple of options:

1. Add the POLARIS subset resource directly to your "modeling project" collection. You wouldn't add it from Nathan Chaney's collection - you would reference the resource you actually used directly in your collection.  Under this scenario, Nathan Chaney's POLARIS **Collection** would not be involved - hence no "collection of collections."
2. Download the POLARIS subset resource, and copy the data into a **Model Instance** resource. If you constructed it this way, it might eventually be executable and there's a higher likelihood that someone might reuse it if they can easily re-execute it. You would describe in the metadata of your model instance resource that you got the soils data from the POLARIS subset resource.  Again - you would reference the POLARIS subset resource you actually used and not Nathan Chaney's POLARIS collection. 

**NOTE**: Unless you construct your resource as a **Model Instance** resource instead of a **Collection** resource, it would not be executable or have any value added functionality. 

**Use Case 2**: I see the utility here, and your description is certainly a valid example of where one might want to create a collection within a collection - i.e., there is one collection for the 2016 student projects, and each project might also be a collection.  But if we could seriously simplify the system by not allowing collections of collections, but rather just tagging each project collection with a keyword of "NFIE 2016" or something like that so they all show up in a search together, you essentially get the same effect.

So, there's ways to support the use cases you've described using existing functionality, but If those aren't the only use cases or we just want to do this to be responsive to the user community by adding the ability to do collections of collections, then we need to consider it.  Ultimately, if the team feels like we should enable collections of collections, we may have to simplify other functionality - like choosing not to add or maintain bi-directional relationships from collections to collected resources - opting instead for simpler one-directional relationships (collections to collected resources), we could still make it work.  My recent suggested simplifications are simply attempts to get us to a set of functionality that is consistent with the resource data model and access control. We've been circling around it for a while and just need to tie it up and move on.

**Comment by ChristinaB** [comment]
I think for Use Case 1, simple works fine.
For Use Case 2, let's sort out the simplest plan for the users and developers. NFIE summer institute is a good use case because it is very similar to academic resaerch groups where multiple projects with various collaborators are coordinated by one group/professor.  

The current plan with existing functionality is to have this years Collection of projects be held under the Collection 2016 Flood inundation and disaster response https://www.hydroshare.org/resource/4a33d8abc8d54b58b6624eda42138d4a/

This Collection will hold Collections of team projects: e.g. FloodHIPPO NFIE 2016
 https://www.hydroshare.org/resource/f643e1039e274818af0923b6054cc752/

The team and summer lead will coordinate to agree on keywords (and other inputs) for each individual resource.  e.g. All summer institute 2016 Keywords: NFIE, flood inundation, disaster response; e.g. FloodHippo project keywords: FloodHIPPO, DisasterZoo, geofencing

The details on how to organize the project are being drafted in this Help resource: https://www.hydroshare.org/resource/2a6612a47f1f40098befc303a3618fe5/
I expect this will be updated regularly over the next two weeks - after which time David Maidment will send out instructions for 30 students to follow suit with uploading their research. 
Whatever comes out of these design conversations, I just need to make the changes to this final destination for user communications. 

My sample of one student (Jim Coll), is that Collections of Collections is good. It gives a visual sense a collective resource being an individual contribution as well as part of a bigger project.  We both like the idea of visualizing the hierarchy using something like this flow chart - would be great if we can automate this.   I can show examples in call tomorrow.  I don't understand the nuances of why Collections of Collections is a bad idea, but am open to letting it go if a simpler workflow can be communicated to users.

**Comment by horsburgh** [comment]
@ChristinaB - the document I shared a link to today allows for collections of collections per your use case. Let's see what people say when they've had a chance to read it.

**Comment by zhiyuli** [comment]
@dtarb @horsburgh @ChristinaB @danames 
We will be implementing the changes to collection in different phases  
The first implementation has been deployed on riverdev.hydroshare.org
which includes:
1) new permission logic:
A resource can be added to a collection if
a) resource is Shareable and user has at least View permission over it  
or
b) user is resource Owner

2) "hasPart" metadata of collection: 
Automatically add hasPart reference whenever a resource is added to a collection, and automatically delete this hasPart reference whenever a resource is removed from a collection.

Independent deletion of a member resource does not automatically propagate to the collection.  Instead when the collection is viewed that resource appears in the deleted contents list which a user with edit permissions can clear.
 
3) "isPartOf" relations of resource: 
dynamically display a "Collected By" list under the "Connections" tab of resource landing page

We have a technical issue in (2): although the "hasPart" metadata automatically gets added or removed when collection changes, and the resourcemetadata.xml gets updated in real time as well, the "Relations" tab can not reflect the changes until user hard-refreshes the landing page. This is due to the limit of currently implemented  display mechanism for most metadata terms, including the "Relations" term.  
 
Another problem is how to deal with existing collection resources on www since they were created before this implementation and have no "hasPart" metadata stored.

**Comment by dtarb** [comment]
Thanks.  @pkdash may be able to give you guidance on writing migration code to add the hasPart relationships to existing collections.  I think that the term "is Collected" might be clearer if it was labeled "Belongs to Collections".  In the Relations tab, consider making the URL a link.  I think it is not a big drawback that content does not update till the page is reloaded.

**Comment by pkdash** [comment]
@zhiyuli The page refresh problem is a general problem for most of the resource types. With a bit of JS code it may not be that hard to solve it in for all resource types. But it's a lower priority issue to address.

**Comment by horsburgh** [comment]
@dtarb, @pkdash, and @zhiyuli - the right semantic relationship type is "hasPart". The way the relation element should work in the Science Metadata is that it stores an RDF triple. So, it should read like a sentence - this resource (the collection resource) "hasPart" <the member resource URL>.

While that's how it should be encoded in the science metadata document because it is RDF/XML, there's nothing saying you can't just display a list of URLs under a header that says something like "Resources that are part of this Collection".

**Comment by zhiyuli** [comment]
@dtarb @horsburgh @danames @pkdash 
If I understood your comments correctly, you would like to keep using "hasPart" term in resourcemetadata.xml to store the relations , but display "hasPart" on landing page in a different way, like showing the url as link, or moving them to a different header.

The issue is: we are re-using existing Relation terms and UI. The value gets written to resourcemetadata.xml and displayed on landing page are exactly the same as the value stored. To display url as link, we need to store html code instead of a plain url. That means the html code will get stored in resourcemetadata.xml as well. Also, when collection is shared around, even the resources that user has no permission will display link (this is not good, and the list on collection landing page already handles this elegantly) .
  
Further, user still can create new "hasPart" terms or others (isCopiedFrom, hostedBy) with arbitrary values, or edit/remove existing "hasPart" terms that were created by system. So it may be difficult to tell which "hasPart" terms are created by system, which "hasPart" terms are created by user, and which system-created "hasPart" terms have been changed by user. That is the pre-condition we can move system-created "hasPart" terms to a different list and give it a different header.

A complete solution would be making a new term and designing new UI for collection for this purpose. But I am not sure it is worthy to do as the list of contained resources on collection landing page already clearly shows most info including relations . And the "hasPart" is already written into resourcemetadata.xml for users who directly work with bag. May we reconsider the necessity of adding more complexity to this 'relation' tab page? Thanks

**Comment by dtarb** [comment]
Given this complexity I am OK if the URL is not displayed as a link.  For a collection we do not want the user to be able to remove the hasPart relationship (except by deleting a member resource).  One way to resolve this is to not show the hasPart relationships that indicate member resources on the relations tab at all.  Then there is no link to display.  The member resources are shown in the collection contents anyway to displaying in the relations tab is repetitive.  hasPart should still be in the science metadata.

**Comment by zhiyuli** [comment]
@dtarb 
Do we keep the ability for a user to directly create "hasPart" metadata? or reserve this term for collection and only system can create/remove it?

**Comment by dtarb** [comment]
I'd be interested in what @horsburgh has to say.  I would think that in the context of a collection it makes no sense for hasPart to refer to anything other than a member resource, so that this should be managed by the system and a user should not be able to directly enter it.

In the context of other resource types I cannot think of a use case where a resource may have a hasPart relationship.  I suppose that the concept could be used in other resource types to include resources by reference, as in a collection of resources that are references within a resource that is a composition.  This would amount to a collection logical "type" that can be within other resources.  Unless and until we develop and think through this concept, I suggest that hasPart not be enabled for non collection resources.

**Comment by ChristinaB** [comment]
Collections should be different than a folder.  If hasPart is enabled only for collections, per David's suggestion, users will use Collections for sharing with the benefit of having the relationship information automatically tracked, which would be great.

**Comment by horsburgh** [comment]
@dtarb and @zhiyuli - I would be fine with removing the "hasPart" term from the relationship drop down that user's can select when they add new relationships. Then, we could use "hasPart" exclusively, and in an automated way for managing collections.

**Comment by horsburgh** [comment]
@dtarb, @zhiyuli, @ChristinaB, @danames, @mauriel133 - after thinking about the implementation of the "Related Resources" tab on the resource landing page, I think maybe our initial implementation is a bit too naive, and perhaps is not serving us as well as it could. The semantic terms "hasPart", "isPartOf", "executedBy", etc. all have meaning that could be interpreted by a computer. But, normal HydroShare users don't necessarily know what that meaning is.  So, we probably shouldn't display relationships like this:

![image](https://cloud.githubusercontent.com/assets/3485902/18258934/b285d3f2-739b-11e6-9778-c64b8a3ee086.png)

Instead, I hope at some point we can be a bit more interpretive and display something intelligible for the user. For example, if a resource is a collection resource and it has several "hasPart" relationships in its metadata, we should interpret that and display something like:

## Resources that are members of this collection resource: ##
Resource1URL
Resource2URL
Resource3URL
...

On the user/display side of things on the resource landing page, the user should probably never see the semantic terms for relationship types.

**Comment by zhiyuli** [comment]
@horsburgh @dtarb @danames 
I agree we may need more work to improve the "Relations" tab for general purpose. Those changes may affect all res types. And it should be addressed in another issue.

For collection res, the member resources are already shown on landing page, so displaying them in the tab again is kind of repetitive. To move on this issue on collection, do you agree to proceed with the above suggestion from @dtarb:   reserve "hasPart" for collection, and hide it on landing page.

**Comment by horsburgh** [comment]
@zhiyuli - I agree that it is repetitive and not needed. Yes - reserve, "hasPart" for collections and remove that term from the list of relationship types that users can select from when editing the "Related Resources" tab

**Comment by zhiyuli** [comment]
@horsburgh 
OK, will get it done. Thanks

@pkdash 
I found I could not add "hasPart" to existing collection resources using migration, which confirmed your guess. 
The error was:
target_res_obj.metadata.create_element("relation", **meta_dict)
AttributeError: 'CollectionResource' object has no attribute 'metadata'

I will try to use django shell then.

**Comment by zhiyuli** [comment]
@dtarb @horsburgh @danames @ChristinaB 

Please check out riverdev.hydroshare.org when you get a chance, and let me know if any other changes needed regarding "hasPart" metadata and "Connections" tab. We ran a script to add "hasPart" metadata to all existing collection resources on riverdev, which set "last change user" to "HydroShare Admin" as  bags were changed.  If everything looks good, the same script will be applied to www once code review is done. Thanks
 
![image](https://cloud.githubusercontent.com/assets/10070599/18292968/616bfc64-744d-11e6-8d33-f9e317d7aaa7.png)

**Comment by horsburgh** [comment]
@zhiyuli - I did some testing. I had a collection resource in my list and it did get modified by your script. I also created a new collection resource and looked through the metadata file. I think the way you encoded the "relation" elements is technically correct. I studied a number of Dublin Core Relation examples (which were hard to find) and came up with a few other options that I don't think were better. I also see that you removed "hasPart" from the list of available relationship types on the "Related Resources" tab.

I didn't see the CSV file listing the collected resources in either collection resource I tested (the one that existed before you rand your script or the one I created after you ran the script). Should it be there?

There's also some weirdness with the BagIt files that I assume is because you deployed this when Hong was in the middle of the fixes to the BagIt stuff.

**Comment by zhiyuli** [comment]
@horsburgh 
CSV file is not implemented yet. Will add it in phase-2.

I am guessing the "weirdness with the BagIt files" you saw was because the branch I am working on does not include Hong's latest code that changed bags, but riverdev is using

**Comment by zhiyuli** [comment]
@horsburgh 
CSV file is not implemented yet. Will add it in phase-2.
What was the "weirdness with the BagIt files" you saw?

**Comment by horsburgh** [comment]
@zhiyuli - it was related to the checksums and checksum algorithms. I don't think it had anything to do with your stuff.

**Comment by pkdash** [comment]
@dtarb Check if this can be closed.
